lib/rex/text.rb

```diff
@@ -1531,7 +1532,8 @@ def self.remove_badchars(data, badchars = '')
   # @param keepers [String]
   # @return [String] All characters not contained in +keepers+
   def self.charset_exclude(keepers)
-    [*(0..255)].pack('C*').delete(keepers)
+    keepers_pat = keepers.unpack("C*").map{|c| "\\x%.2x" % c}.join
+    [*(0..255)].pack("C*").gsub(/[^#{keepers_pat}]/n, '')
```

Comment:
Since you're already treating these as bytes instead of characters, it's easier to just delete them from the range/array, with something like this:
```ruby
def self.charset_exclude(keepers)
  excluded_bytes = [*(0..255)] - keepers.unpack("C*")
  excluded_bytes.pack("C*")
end
```